src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -351,11 +420,37 @@ export class SavedObjectsClient {
     return resolveUrl(API_BASE_URL, join(...path));
   }
 
-  private request({ method, path, query, body }: RequestParams) {
-    if (method === 'GET' && body) {
-      return Promise.reject(new Error('body not permitted for GET requests'));
-    }
-
-    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) });
+  /**
+   * To ensure we don't break backwards compatibility, savedObjectsFetch keeps
+   * the old kfetch error format of `{res: {status: number}}` whereas `http.fetch`
+   * uses `{response: {status: number}}`
+   */
+  private savedObjectsFetch(path: string, { method, query, body }: HttpFetchOptions) {
+    return this.http.fetch(path, { method, query, body }).catch(err => {
+      const kfetchError = Object.assign(err, { res: err.response });
+      return Promise.reject(kfetchError);
+    });
   }
 }
+
+/**
+ * Returns a new object with the own properties of `obj`, but the
+ * keys renamed according to the `keysMap`.
+ *
+ * If a key is not found in `keysMap` it is omitted from the result.
+ *
+ * @param keysMap - a map of the form `{oldKey: newKey}`
+ * @param obj - the object whose own properties will be renamed
+ */
+const pickAndRenameKeys = <T extends Record<string, any>, U extends Record<string, any>>(
+  keysMap: Record<keyof T, keyof U>,
+  obj: Record<string, any>
+) =>
+  Object.keys(obj).reduce((acc, key) => {
```

Comment:
May want to use `getOwnPropertyNames` instead of `keys` to avoid prototype pollution.

Reply:
> The Object.keys() method returns an array of a given object's own property names
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys